JobRunner/JobRunner.py

```diff
@@ -197,6 +205,16 @@ def _validate_token(self):
 
         return user
 
+    def _get_token_lifetime(self, config):
+        try:
```

Comment:
Do we really want to fail because we cannot get a response from AUTH? Auth goes down sometimes and this would cause this job to fail. We might want to wait and retry a few more minutes before causing this failure.  An alternative implementation is to start a death timer to automatically kill the job when the token expires. Currently these tokens being passed in are agent tokens which cannot be easily revoked by the user AFAIK.

Reply:
Sorry I was wrong, I thought this code is executed multiple times. However, it might still be good to do a timeout and retry for this.